pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/codestyle/xml/MethodNamingConventions.xml

```diff
@@ -66,4 +66,16 @@ public class Foo {
         ]]></code>
     </test-code>
     
+    <test-code>
+        <description>#1573 method names should not contain underscores, but skip test methods</description>
+        <expected-problems>0</expected-problems>
+        <rule-property name="skipTestMethodUnderscores">true</rule-property>
```

Comment:
we are missing an equivalent test with this property set to false, and checking we get 1 violation reported

Reply:
Thanks for the comments - should I add three more testcases to represent the different test annotations?
1 (current test case) - @isTest, property true, 0 expected problems
2 - @isTest, property false, 1 expected problem
3 - testMethod property true, 0 expected problems
4 - testMethod, property false, 1 expected problem

or would 1, 2 and 3 be enough to cover the bases?

Reply:
Having the 4 would be most complete, but 3 is probably good enough for coverage